src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -630,7 +650,7 @@ ResponseCode RTCPMessageManager::processKeepAliveResponse(
 }
 
 ResponseCode RTCPMessageManager::processRTCPMessage(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.